File: Modules/SEDO/EMS/SIE.Web.EMS/Checks/Plans/Commands/GenerateSparePartAppCommand.js
SIE.defineCommand('SIE.Web.EMS.Checks.Plans.Commands.GenerateSparePartAppCommand', {
    meta: { text: "申请", group: "edit", iconCls: "icon-Check" },
    canExecute: function (view) {
        var entity = view.getParent().getCurrent();
        return (entity.getExeState() == 0 || entity.getExeState() == 4);
    },
    execute: function (view) {
        var commandInfo = {
            command: "SIE.Web.EMS.Checks.Plans.Commands.GenerateSparePartAppCommand",
            entityType: "SIE.EMS.Checks.Plans.CheckPlanSparePartApl",
            parentType: "SIE.EMS.Checks.Plans.ViewModels.CheckPlanViewModel",
            moduleName: "点检计划维护",
            childModuleName: "",
            commandName: "申请",
        }
        var datas = [];
        Ext.each(view.getData().data.items, function (data) { datas.push(data.data); });
        view.execute({
            data: datas,
            logInfo: commandInfo,
            success: function (res) {
                SIE.Msg.showInstantMessage('备件申请成功'.t());
                view.loadChildData(true);
            }
        });
    }
});

File: Modules/SEDO/EMS/SIE.Web.EMS/Checks/Plans/Commands/GenerateSparePartAppCommand.cs
using SIE.Domain;
using SIE.Domain.Validation;
using SIE.EMS.Checks.Plans;
using SIE.EMS.SpareParts.Applys.Controllers;
using SIE.Web.Command;
using System;
using System.Collections.Generic;
using System.Linq;

namespace SIE.Web.EMS.Checks.Plans.Commands
{
    /// <summary>
    /// 生成备件申请单 命令
    /// </summary>
    [JsCommand("SIE.Web.EMS.Checks.Plans.Commands.GenerateSparePartAppCommand")]
    public class GenerateSparePartAppCommand : ViewCommand
    {
        /// <summary>
        /// 执行生成备件申请单命令
        /// </summary>
        /// <param name="args">args</param>
        /// <param name="scope">scope</param>
        /// <returns>执行结果</returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            var datas = args.Data.ToJsonObject<List<CheckPlanSparePartApl>>();
            if (datas.Count > 0)
                RT.Service.Resolve<SparePartAppController>().UIGenerateSparePartApp(datas);
            else
                throw new ValidationException("未添加申请单明细".L10N());

            return true;
        }
    }
}
